codersdk/workspacesdk/dialer.go

```diff
@@ -24,6 +24,8 @@ var permanentErrorStatuses = []int{
 	http.StatusBadRequest,          // returned if API mismatch
 	http.StatusNotFound,            // returned if user doesn't have permission or agent doesn't exist
 	http.StatusInternalServerError, // returned if database is not reachable,
+	http.StatusUnauthorized,        // returned if user is not authenticated
```

Comment:
You should include a comment above whichever http status the resume_token error sets saying that it's conditional